### PR TITLE
Support in place transform

### PR DESCRIPTION
**Pull request by LucienLee**
Feat #301 Support transform test file as default behaviour.

- Make in place transform as default and use `.transformed` suffix.
- Provide `outputDir` option for specifying directory.
- Use `poi.argv.baseDir`as base directory to glob test file.

### CODE REVIEW

**Comment by LucienLee** [comment]
@egoist Anything need to polish? 😃

File: packages/poi-preset-transform-test-files/index.js [hunk 1]
```diff
@@ -2,27 +2,37 @@ const path = require('path')
...
       const ignores = ['!**/node_modules/**', '!**/vendor/**']
 
-      return globby(inputFiles.concat(ignores), { cwd: poi.options.cwd })
+      return globby(inputFiles.concat(ignores), { cwd: baseDir })
         .then(files => {
           delete webpackConfig.entry.client
-          webpackConfig.entry.test = files
-            .map(file => path.resolve(poi.options.cwd, file))
+          webpackConfig.entry = files.reduce((acc, file) => {
+            const filename = outputDir ? path.basename(file) : file
```
Comment (egoist) [inline packages/poi-preset-transform-test-files/index.js]: why `path.basename`? seems `test/nest/test.js` and `test/test.js` will end up being transformed to the same location which is `outputDir/test.transformed.js`

Comment (LucienLee) [reply packages/poi-preset-transform-test-files/index.js]: In my test, the location would be different as default when `outputDir` is empty.

The spec is: 
If users not specify `outputDir`, the `test/nest/test.js` and `test/test.js` will transform as `test/nest/test. transformed.js` and `test/test.transformed.js`. However, if `outputDir` is not '', they would be put into `outputDir/...` centrally. 

The reason is that Jest encourage developers put test file around source file under `__test__`, which is dispersed. In order to match the pattern, I set in place transform as default behaviour.

**Comment by LucienLee** [comment]
Looks great!

**Pull request merged**